### PR TITLE
Add support for v2 parameters

### PR DESCRIPTION
**Pull request by mogproject**

### CODE REVIEW

File: src/exp_runmemtests.py [hunk 1]
```diff
@@ -373,33 +375,60 @@ def main():
...
     tf_pp_dirname = 'data/post_preprocessing/tf/'
     lv_pp_dirname = 'data/post_preprocessing/lv/'
     
-    tf_out_dirname = 'data/finaldata/tf/'
-    lv_out_dirname = 'data/finaldata/lv/'
+    out_dir_suffix = {
+        (True, True): '_kv2_pv2',
+        (True, False): '_kv2_pv1',
+        (False, True): '_kv1_pv2',
+        (False, False): '',
```
Comment (sjain12) [inline src/exp_runmemtests.py:415]: Perhaps we should add a separate directory for the _kv1_pv1 case also, to keep it cleaner?

Comment (mogproject) [reply src/exp_runmemtests.py:415]: This is for backward compatibility. We could rename the directory in the analysis phase if we want to.

**Comment by mogproject** [comment]
Example output with `num_total_blocks`

```{'true_total_kernel': {'passed_kernel': True, 'A_kernel':      112  384  227  210  231  54
112  inf  2.0  4.0  4.0  4.0  2.0
384  2.0  2.0  2.0  2.0  2.0  0.0
227  4.0  2.0  inf  4.0  4.0  2.0
210  4.0  2.0  4.0  inf  4.0  2.0
231  4.0  2.0  4.0  4.0  inf  2.0
54   2.0  0.0  2.0  2.0  2.0  2.0, 'removal_vertices': [385, 386, ..., 255], **'num_total_blocks': 3**, 'num_reduced_blocks': 2, 'clique_vertices': [[54, 112, 210, 227, 231], [112, 210, 227, 231, 384]], 'n': 6, 'm': 14, 'max_eweight': 4.0, 'w_inf': 16.0, 'kinput': 4, 'kernel_time': 0.786733865737915}, 'true_distinct_kernel': {'passed_kernel': True, 'A_kernel':      112  384  227  210  231  54
112  inf  2.0  4.0  4.0  4.0  2.0
384  2.0  2.0  2.0  2.0  2.0  0.0
227  4.0  2.0  inf  4.0  4.0  2.0
210  4.0  2.0  4.0  inf  4.0  2.0
231  4.0  2.0  4.0  4.0  inf  2.0
54   2.0  0.0  2.0  2.0  2.0  2.0, 'removal_vertices': [385, 386, ..., 255], **'num_total_blocks': 3**, 'num_reduced_blocks': 2, 'clique_vertices': [[54, 112, 210, 227, 231], [112, 210, 227, 231, 384]], 'n': 6, 'm': 14, 'max_eweight': 4.0, 'w_inf': 16.0, 'kinput': 2, 'kernel_time': 0.752100944519043}, 'guess_kernels': {-0.6: None, -0.4: None, -0.2: None, 0.2: None, 0.4: None, 0.6: None}}
```

File: src/main.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def run_bsd(A, G, k_input, alg_version, perf_lp_v2_enabled, num_threads):
         B = bsd_dw.BSD_DW(A, k_input, winf) 
     elif alg_version == 'lp':  
         # linear programming version
-        B, W = bswd_dw_lp.BSWD_DW(A, k_input, perf_lp_v2_enabled, num_threads)
+        B, W, num_lp_runs = bswd_dw_lp.BSWD_DW(A, k_input, perf_lp_v2_enabled, num_threads)
```
Comment (sjain12) [inline src/main.py:120]: Minor comment: num_lp_runs is being collected here but not returned along with B and W. Won't make a difference for our experiments though.

**Comment by sjain12** [comment]
Included a couple of minor comments. Looks good otherwise.

File: src/main.py (hunk 2 above)
Comment (mogproject) [reply src/main.py:120]: Yes, this is on purpose.

**Pull request merged**